src/meta/src/rpc/service/scale_service.rs

```diff
@@ -132,7 +137,19 @@ where
     ) -> Result<Response<RescheduleResponse>, Status> {
         let req = request.into_inner();
 
-        self.stream_manager
+        let _guard = self.stream_manager.reschedule_revision_lock.lock().await;
+
+        let current_revision = self.stream_manager.reschedule_revision().await?;
+
+        if req.revision != current_revision.inner() {
```

Comment:
Should error be returned to give a clear error message?

Reply:
Or let's rename "success" as "revision_up_to_date"? 😄

Reply:
How about using this "oneof" to represent the return status? 🤔 
```message RescheduleResponse {
  message RevisionMismatch {
    uint64 expected_revision = 1;
    uint64 actual_revision = 2;
  }
  message Success {
    uint64 revision = 1;
  }
  oneof result {
    Success success = 1;
    RevisionMismatch revision_mismatch = 2;
  }
}```